### PR TITLE
notifier: use testutil for testing

### PR DESCRIPTION
**Pull request by smonv**
- Use `util/testutil` in `notifier` testsuit. Related #3242

### CODE REVIEW

File: notifier/notifier_test.go [hunk 1]
```diff
@@ -80,34 +79,23 @@ func TestHandlerNextBatch(t *testing.T) {
...
 
-	if len(b) != maxBatchSize {
-		t.Fatalf("Expected first batch of length %d, but got %d", maxBatchSize, len(b))
-	}
-	if !alertsEqual(expected[0:maxBatchSize], b) {
-		t.Fatalf("First batch did not match")
-	}
+	testutil.Equals(t, maxBatchSize, len(b))
+
+	testutil.Assert(t, alertsEqual(expected[0:maxBatchSize], b), "First batch did not match")
```
Comment (cstyan) [inline notifier/notifier_test.go]: maybe `alertsEqual` is something we want in `testutil?

Comment (krasi-georgiev) [reply notifier/notifier_test.go]: Yep , that sound reasonable.

**Comment by tthanh** [comment]
thanks for feedback, I'll update this PR

File: notifier/notifier_test.go (hunk 1 above)
Comment (tthanh) [reply notifier/notifier_test.go]: I think we can move the check len of 2 alert array outsite of `alertEqual` so `alertEqual` only contain the for loop.

**Comment by tthanh** [comment]
@cstyan @krasi-georgiev I won't move `alertEqual` to `testutil` package because it specific to the alerts. Can you help provide more idea which part of `alertEqual` should I improve.

Updated resolve conflicts of `notifier` package with master

**Comment by krasi-georgiev** [comment]
Did you check if you can refactor these to make it table based.
I find it more readable and easy to understand.

something like:
https://github.com/prometheus/prometheus/blob/6b0e9ef183fddf5b27b4ea4d2052153a6bdf0660/notifier/notifier_test.go#L41-L71

**Comment by tthanh** [comment]
@krasi-georgiev thanks for feedback. I'll check again and update.

**Comment by smonv** [comment]
@krasi-georgiev After check I think I can't not make more test like the table you mentioned. The test was step by step check, identical test flow, not multiple case.

File: notifier/notifier_test.go [hunk 2]
```diff
@@ -443,13 +411,10 @@ func (a alertmanagerMock) url() *url.URL {
...
-	if err != nil {
-		t.Fatal(err)
-	}
 
-	if !reflect.DeepEqual(tg, makeInputTargetGroup()) {
-		t.Fatal("Target modified during alertmanager extraction")
-	}
+	testutil.Ok(t, err)
+
+	testutil.Assert(t, reflect.DeepEqual(tg, makeInputTargetGroup()), "Target modified during alertmanager extraction")
```
Comment (brian-brazil) [inline notifier/notifier_test.go]: Can you not use testutil.Equals here?

Comment (smonv) [reply notifier/notifier_test.go]: I didn't use `testutil.Equals` because it can not print custom message and I want respect original fatal message.

Comment (brian-brazil) [reply notifier/notifier_test.go]: That doesn't give us the diff then. I'd suggest making the message a comment.

Comment (smonv) [reply notifier/notifier_test.go]: Ok, I'll update it to `testutil.Equals`. Thank for feedback

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**